velox/vector/tests/VectorTest.cpp

```diff
@@ -576,6 +577,99 @@ class VectorTest : public testing::Test, public test::VectorTestBase {
     testCopy(lazy, level - 1);
   }
 
+  static void testSlice(
+      const VectorPtr& vec,
+      int level,
+      vector_size_t offset,
+      vector_size_t length) {
+    SCOPED_TRACE(fmt::format(
+        "testSlice encoding={} offset={} length={}",
+        vec->encoding(),
+        offset,
+        length));
+    ASSERT_GE(vec->size(), offset + length);
+    auto slice = vec->loadedVector()->slice(offset, length);
+    ASSERT_EQ(slice->size(), length);
+    // Compare values and nulls directly.
+    for (int i = 0; i < length; ++i) {
+      EXPECT_TRUE(slice->equalValueAt(vec.get(), i, offset + i));
+    }
```

Comment:
nit: maybe we can break this down into smaller sub tests; so if things fail, we know exactly what functionality.

Reply:
It's hard to do so because tests are organized by vector types.  I put enough information with `SCOPED_TRACE` though, so we know exact what kind of encodings are having trouble.